cdk/HOWTO.md

```diff
@@ -0,0 +1,432 @@
+</p>
+
+4. Click `Save new version` button and `Start deployment` button.
+
+5. Select deployment method and deploy.
+
+6. Check `{backend_url}/feature-flags` endpoint on backend application.
+
+---
+
+## Add new feature flag
+
+1. Add a new feature flag key and value (True/False) to https://github.com/IGVF-DACC/igvfd/blob/dev/cdk/infrastructure/config.py for a specific environment:
+
+<p align="center">
+  <img src="./images/howto/add-new-feature-flag_1.png" alt="Add new feature flag 1" width="500">
+</p>
+
+2. Make sure all flags in the config are set to desired state for next flag deployment (adding or changing any flag in the config will cause a new deployment of the flags, overwritting any flags toggled in the console).
+
+3. Commit code and push.
+
+---
+
+## Connect to Postgres with psql
+
+1. Log into running backend application container using [Log into running container](#log-into-running-container).
+
+2. Install `psql` with `# apt-get update && apt-get install postgresql-client`.
+
+3. Create database url with `# export SQLALCHEMY_URL=postgresql://postgres:${DB_PASSWORD}@${DB_HOST}/${DB_NAME}`.
+
+4. Connect to database with `# psql $SQLALCHEMY_URL`.
+
+---
+
+## Connect to Opensearch
+
+1. Log into running backend application container using [Log into running container](#log-into-running-container).
+
+2. Query with cURL or Python client: `# curl -XGET ${OPENSEARCH_URL}/_cat/indices`.
+
+---
+
+## Restore database snapshot to lower environments
+
+1. Delete `Backend`, `Opensearch`, and `Postgres` stacks in CloudFormation console.
+
+2. (Optional) Update `postgres` config in https://github.com/IGVF-DACC/igvfd/blob/dev/cdk/infrastructure/config.py for a specific environment with a new `snapshot_arn` or `snapshot_source_db_identifier` (or leave the same if you want latest snapshot from same source).
+
+3. Commit code and push if (2) or trigger new deployment manually with [Retrigger release pipeline](#retrigger-release-pipeline).
+
+---
+
+## Swap Opensearch
+
+(Try swapping on demo instance first.)
+
+1. Specify a new Opensearch cluster in https://github.com/IGVF-DACC/igvfd/blob/dev/cdk/infrastructure/config.py for a specific environment:
+
+   a. Use a unique `construct_id` name (e.g. `Opensearch27`).
+
+   b. Update the `backend` config to `write_to_opensearch_named` with the new `construct_id` name  (e.g. `Opensearch27`).
+
+<p align="center">
+  <img src="./images/howto/swap-opensearch_1.png" alt="Swap Opensearch 1" width="500">
+</p>
+
+2. Commit, push, and deploy changes through all environments (dev, staging, sandbox, prod). For each environment:
+
+   a. Index everything into the new cluster with [Trigger index creation](#trigger-index-creation) and [Trigger reindexing](#trigger-reindexing).
+
+   b. Wait for indexing to finish (monitor indexing progress by number of documents metric in [Check Opensearch logs](#check-opensearch-logs)).
+
+   c. Promote change to next environment and repeat (a) and (b).
```

Comment:
Are there some specific steps that one needs to do to 'Promote change to the next environment' or does it simply mean 'Repeat (a) and (b) in next enviroment'? In that case maybe would just change it into latter.